collector/generatorreceiver/generator_receiver.go

```diff
@@ -209,3 +215,29 @@ func newTraceReceiver(config *Config,
 	genReceiver.traceConsumer = consumer
 	return &genReceiver, nil
 }
+
+func validateConfiguration(topoFile topology.File) error {
+	err := flags.Manager.ValidateFlags()
+	if err != nil {
+		return fmt.Errorf("validation of flag configuration failed: %v", err)
+	}
+
+	for _, service := range topoFile.Topology.Services {
+		err = service.Validate(*topoFile.Topology)
+		if err != nil {
+			return fmt.Errorf("validation of service configuration failed: %v", err)
+		}
+	}
+
+	err = topoFile.Topology.ValidateServiceGraph(topoFile.RootRoutes) // depends on all services/routes being validated (i.e. exist) first ^
+	if err != nil {
+		return fmt.Errorf("cyclical service graph detected: %v", err)
+	}
+
+	err = topoFile.ValidateRootRoutes()
```

Comment:
Oops now that I pass `RootRoutes` to `ValidateServiceGraph()` I should call `ValidateRootRoutes()` before it.